psi4/src/psi4/libmints/integral.cc

```diff
@@ -113,11 +115,21 @@ OneBodySOInt* IntegralFactory::so_potential(int deriv) {
     return new PotentialSOInt(ao_int, this);
 }
 
-OneBodyAOInt* IntegralFactory::ao_ecp(int deriv) { return new ECPInt(spherical_transforms_, bs1_, bs2_, deriv); }
+OneBodyAOInt* IntegralFactory::ao_ecp(int deriv) {
```

Comment:
Maybe in the future these could return unique_ptrs directly.

Reply:
There's already an issue about this: https://github.com/psi4/psi4/issues/2493